src/SQL/Parser.php

```diff
@@ -34,13 +34,13 @@
  */
 final class Parser
 {
-    private const SPECIAL_CHARS = ':\?\'"`\\[\\-\\/';
+    private const SPECIAL_CHARS = ':\?\'"`\\[\\-\\/$';
 
     private const BACKTICK_IDENTIFIER  = '`[^`]*`';
     private const BRACKET_IDENTIFIER   = '(?<!\b(?i:ARRAY))\[(?:[^\]])*\]';
     private const MULTICHAR            = ':{2,}';
     private const NAMED_PARAMETER      = ':[a-zA-Z0-9_]+';
-    private const POSITIONAL_PARAMETER = '(?<!\\?)\\?(?!\\?)';
+    private const POSITIONAL_PARAMETER = '((?<!\\?)\\?(?!\\?)|\\$\d+)';
```

Comment:
Would it make sense to extract the new pattern into a separate constant?

Reply:
I don't think so. Two arguments:
- This is Postgres' way of specifying positional parameters. So the naming makes a lot of sense.
- Adding an additional pattern would incur a performance penalty and complicate the code (having to add one more handler, which in turn just calls $visitor->acceptPositionalParameter($sql);)